app/assets/javascripts/components/common/ArticleViewer/containers/ArticleViewer.jsx

```diff
@@ -170,11 +171,31 @@ const ArticleViewer = ({ showOnMount, users, showArticleFinder, showButtonLabel,
       const styledAuthorSpan = `<span title="${user.name}" class="editor-token token-editor-${user.userid} ${colorClass}`;
       const authorSpanMatcher = new RegExp(`<span class="editor-token token-editor-${user.userid}`, 'g');
       html = html.replace(authorSpanMatcher, styledAuthorSpan);
-      if (prevHtml !== html) user.activeRevision = true;
+      if (prevHtml !== html) {
+          user.activeRevision = true;
+      } else {
+        // If highlighting failed, verify if the user's has contributions in the article using checkFallback.
+        checkFallback(user);
+      }
     });
     setHighlightedHtml(html);
   };
 
+  const checkFallback = (user) => {
```

Comment:
As currently implemented, it looks like this would be run multiple times if multiple users edited the article but had no highlighted contributions. The wikitext metadata should only need to be fetched once at most.

Reply:
This function name is kind of vague. Can you think of a more precise one? A comment explaining the purpose of this would also be useful.